tasks/main.yml

```diff
@@ -2,10 +2,15 @@
 - name: Include OS-specific variables.
   include_vars: "{{ ansible_os_family }}.yml"
 
+- name: Check Xdebug is installed.
+  stat: path={{ workspace }}/xdebug-{{ php_xdebug_version }}/modules/xdebug.so
+  register: xdebug_module
+
 - name: Download Xdebug.
   get_url:
     url: "http://xdebug.org/files/xdebug-{{ php_xdebug_version }}.tgz"
     dest: "{{ workspace }}"
+  when: not xdebug_module.stat.exists
```

Comment:
There's a simpler way to do this using `get_url`—if you specify the path to the _file_ instead of the directory, Ansible won't re-download the file every time the role is run. This can be done by changing `dest` to `"{{ workspace }}/xdebug-{{ php_xdebug_version }}.tgz", I think.

Can you change that and then we'll see if the role works the same (and doesn't re-download the file)? This would be simpler and make the role slightly less brittle, imo.